### PR TITLE
Add Z60 PCB Solder version

### PR DESCRIPTION
**Pull request by tommyamoszhao**
Variants:
Split backspace, nor/stepped capslock, split left shift, split right shift, ISO enter Bottom row variants:
6.25U spacebar, 7U spacebar, HHKB and WKL

## Types of Changes
- [ Add ] Z60 Keyboard solder version
- [x] Via enabled

## Checklist

- [ * ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ * ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ * ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ * ] I have added tests to cover my changes.
- [ * ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/quarkeys/z60/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		
+	}
+
+	if (usb_led & (1 << USB_LED_KANA)) {
+		
+	} else {
+		
+	}
+
+}
```
Comment (waffle87) [inline keyboards/quarkeys/z60/keymaps/default/keymap.c:112]: The indicator functionality belongs in `config.h`: [https://docs.qmk.fm/#/feature_led_indicators](https://docs.qmk.fm/#/feature_led_indicators).
```suggestion
```

File: keyboards/quarkeys/z60/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/quarkeys/z60/keymaps/via/keymap.c:1]: This file is missing a license header.

File: keyboards/quarkeys/z60/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+	[0] = LAYOUT_all(
+		KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL, 
+		LT(3, KC_TAB),    KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, 
+		LT(4, KC_CAPS),   KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_HASH, KC_ENT, 
+		KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, RSFT_T(KC_UP),    MO(1), 
+		KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,  MO(2),   RALT_T(KC_LEFT),  RGUI_T(KC_DOWN),  RCTL_T(KC_RGHT)),
+
+	[1] = LAYOUT_all( 
+		RESET,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS, KC_TRNS,
```
Comment (waffle87) [inline keyboards/quarkeys/z60/keymaps/via/keymap.c:13]:
```suggestion
		QK_BOOT,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS, KC_TRNS, 
```

File: keyboards/quarkeys/z60/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		
+	}
+
+	if (usb_led & (1 << USB_LED_KANA)) {
+		
+	} else {
+		
+	}
+
+}
```
Comment (waffle87) [inline keyboards/quarkeys/z60/keymaps/via/keymap.c:112]:
```suggestion
```

File: keyboards/quarkeys/z60/rules.mk [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+#F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+#OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+#OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (waffle87) [inline keyboards/quarkeys/z60/rules.mk:43]:
```suggestion
# Bootloader selection
BOOTLOADER = atmel-dfu
```

File: keyboards/quarkeys/z60/z60.c [hunk 6]
```diff
@@ -0,0 +1 @@
+#include "z60.h"
```
Comment (waffle87) [inline keyboards/quarkeys/z60/z60.c:1]: This file is missing a license header.

File: keyboards/quarkeys/z60/z60.h [hunk 7]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+#ifndef Z60_H
+#define Z60_H
+
```
Comment (waffle87) [inline keyboards/quarkeys/z60/z60.h:4]: This file is missing a license header.
```suggestion
```

File: keyboards/quarkeys/z60/z60.h [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+	K400, K401,       K403,             K406,                   K410, K411, K412, K413  						\
+)  { \
+	{ K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013,  K014 }, \
+	{ XXXX,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111,  K112,  K113,  K114 }, \
+	{ XXXX,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  K212,  K213,  K214 }, \
+	{ K300,  K301,  K302,  K303,  K304,  K305,  K306,  K307,  K308,  K309,  K310,  K311,  K312,  K313,  XXXX }, \
+	{ K400,  K401,  XXXX,  K403,  XXXX,  XXXX,  K406,  XXXX,  XXXX,  XXXX,  K410,  K411,  K412,  K413,  XXXX }  \
+}
+
+#endif
```
Comment (waffle87) [inline keyboards/quarkeys/z60/z60.h:22]:
```suggestion
```

File: keyboards/quarkeys/z60/config.h [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+#define RGBLIGHT_ANIMATIONS
+#define RGB_DI_PIN D6
+#define RGBLED_NUM 77
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+
+#define RGBLIGHT_LAYERS
+#define RGBLIGHT_LAYERS_OVERRIDE_RGB_OFF
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true
```
Comment (waffle87) [inline keyboards/quarkeys/z60/config.h:30]:
```suggestion
#define RGBLIGHT_SLEEP 
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by tommyamoszhao** [comment]
Hi Jack,
This is TommyZ from Quarkeys studio. I couldn’t thank you enough for the much detailed and in time message, the QMK I’ve been using turned out to be an older version, didn’t realize until I send the pull request. Sorry for taking much of your time to correct this. One question though, when I upload quarkeys/z60 firmware file, it shows on GitHub with “quarkeys/z60”, actually, there are more keyboards other than z60, so is there any way that I could make the directory on qmk_firmware:master shows only “quarkeys” with many subdirectory of “Z60” and “Z40” so on? Or, it is an automatic process after other keyboards files are updated? Cheers.
Sincerely, TommyZ

**Comment by waffle87** [comment]
@tommyamoszhao You can add multiple keyboards to the `quarkeys/` directory — it's a common practice throughout the repo. 
eg.
```
quarkeys
├── z40
├── z60
└── z_etc

3 directories
```
However, please add additional keyboards in separate pull requests.

**Pull request closed without merging**